### ISSUE TITLE
`entrypointTypes` doesn't seem to do anything

### ISSUE THREAD

**Issue by RaptorKyelok** [projen/projen#2619]
Adding the `entrypointTypes` parameter to my project options isn't changing the `types` output in `package.json`.

`entrypoint` does, however, work to change the `main` parameter.

**Comment by mrgrain** [comment]
Thanks for reporting this. We don't have test coverage for this at the moment, so I can't tell for sure what's up.

The code seems to be there in principle though:

```ts
    if (options.entrypointTypes || this.entrypoint !== "") {
      const entrypointTypes =
        options.entrypointTypes ??
        `${path
          .join(
            path.dirname(this.entrypoint),
            path.basename(this.entrypoint, ".js")
          )
          .replace(/\\/g, "/")}.d.ts`;
      this.package.addField("types", entrypointTypes);
    }
```

**Comment by RaptorKyelok** [comment]
Yep, I saw that. Interestingly enough, changing `entrypoint` changes `main` but not `types`, while this code is supposed to change `types` based on `entrypoint` if it was given. Suggests to me that this code doesn't set the parameter properly.

`entrypoint` is set in a different path, via ... `NodeProject`'s constructor?

**Comment by RaptorKyelok** [comment]
Adding `project.package.addFields("types", "my-types.d.ts")` in `.projenrc.ts` works to change the `types` parameter. Hmm.

**Comment by RaptorKyelok** [comment]
Found it: https://github.com/projen/projen/blob/41c719f00fb7bd9ee9c9cbd06ad31a3dd5f047c5/src/cdk/jsii-project.ts#L222

**Comment by RaptorKyelok** [comment]
This overwrites what is set in the `TypescriptProject` constructor.

**Comment by mrgrain** [comment]
Ah, so it's a JsiiProject issue. Thanks for investigating. 😃

**Comment by mrgrain** [comment]
small bug in JsiiProject which wants to force set this field (not sure it should).
